### PR TITLE
Adding alert support for BigQuery plugin

### PR DESCRIPTION
**Pull request by LiorRacer**
Adding alert support for BigQuery plugin

Update BigQuery data source to 2.0.0

### CODE REVIEW

**Comment by daniellee** [comment]
@LiorRacer I'm getting errors when loading the backend component on the latest Grafana master. Which version of Grafana is this compatible with?

You should add this information to the [plugin.json file](https://grafana.com/docs/grafana/latest/developers/plugins/metadata/#properties-1) as it definitely won't work with older versions of Grafana.

**Comment by LiorRacer** [comment]
Hi @daniellee 
I'm using version 7.0.3,  
I will add it to plugin.json file under `grafanaVersion`

Let me know if you need anything else

**Comment by LiorRacer** [comment]
Hi @daniellee 
Is there any news regarding signing a plugin?

**Comment by Tamirk** [comment]
Hi @daniellee
Wanted to follow on this as well and understand when can we expect this to be merged and sign and be available to the public community?

**Comment by marcusolsson** [comment]
Thank you for your patience while waiting for us to review your plugin. 

We're working hard at improving the plugin experience for both users and authors. I'd like to share some important information on how to package and publish plugins going forward.

Since Grafana 7.3, we require all plugins to be _signed_. By signing your plugin, users can guarantee the integrity of the plugin and that the plugin version they've installed hasn't been modified since it was released. For more information on how to sign your plugin, refer to [Sign a plugin](https://grafana.com/docs/grafana/latest/developers/plugins/sign-a-plugin/).

With this, we now require plugins to be _packaged_ before we publish them. This means that instead of a commit containing a `dist` directory, you will need to submit a link to a ZIP archive containing your plugin. For more information on how to package your plugin, refer to [Package a plugin](https://grafana.com/docs/grafana/latest/developers/plugins/package-a-plugin/).

Once you've packaged your plugin, you'll need to update your entry to `repo.json` to the new format. You can read more on how to do this under [Add a plugin to repo.json](https://github.com/grafana/grafana-plugin-repository#add-a-plugin-to-repojson).

To summarize, to publish your plugin, make sure that you:

1. [Sign your plugin](https://grafana.com/docs/grafana/latest/developers/plugins/sign-a-plugin/)
2. [Package your plugin](https://grafana.com/docs/grafana/latest/developers/plugins/package-a-plugin/)
3. Create a new GitHub release and upload the packaged plugin as a release asset
4. [Update your repo.json entry](https://github.com/grafana/grafana-plugin-repository#add-a-plugin-to-repojson)

Here's an [example](https://github.com/marcusolsson/grafana-jsonapi-datasource/releases/tag/v0.6.0) of a properly packaged plugin. In the same repository, you can find a [GitHub Action](https://github.com/marcusolsson/grafana-jsonapi-datasource/blob/master/.github/workflows/main.yml) that signs, packages, and releases the plugin whenever you create a new tag.

Once you've created a GitHub release with the packaged plugin, you can right-click the release asset, copy the link, and validate it using [the plugin validator](https://grafana-plugins-web-vgmmyppaka-lz.a.run.app). 

Finally, in the November [community call](https://groups.google.com/g/grafana-developers/c/ps2vzCXRDCM/m/i7uAK1u9AQAJ), I'll be walking through the steps needed to sign and package your plugin. A recording will be made available on our [YouTube account](https://www.youtube.com/playlist?list=PLDGkOdUX1Ujo0OyE5HFffHE9GJVvaVXGs), in case you can't make it.

**Comment by marcusolsson** [comment]
Hi @LiorRacer,

The linter checks doesn't seem to be related to this plugin so feel free to ignore it for now.

However, could you please update the PR according to the [new format](https://github.com/grafana/grafana-plugin-repository#add-a-plugin-to-repojson):

Example:

```
{
  "id": "briangann-gauge-panel",
  "type": "panel",
  "url": "https://github.com/briangann/grafana-gauge-panel",
  "versions": [
    {
      "version": "0.0.8",
      "url": "https://github.com/briangann/grafana-gauge-panel",
      "download": {
        "any": {
          "url": "https://github.com/briangann/grafana-gauge-panel/releases/download/v0.0.8/briangann-gauge-panel-0.0.8.zip",
          "md5": "782c973460f330287b7efca5486aa015"
        }
      }
    }
  ]
}
```

**Comment by marcusolsson** [comment]
The URL in the PR seems to have a typo. I assume this is the correct one:

https://github.com/doitintl/bigquery-grafana/releases/download/2.0.1/doitintl-bigquery-datasource-2.0.1.zip

Even with the correct URL, I'm getting a few issues related to an unsupported archive structure:

Screen Shot 2021-01-07 at 09 03 21

Could you make sure that you're following the instructions on [Package a plugin](https://grafana.com/docs/grafana/latest/developers/plugins/package-a-plugin/)?

```
mv dist/ doitintl-bigquery-datasource
zip doitintl-bigquery-datasource-2.0.1.zip doitintl-bigquery-datasource -r
```

**Comment by marcusolsson** [comment]
Thank you! I've moved this to Waiting for Review. You can follow the progress on the [project board](https://github.com/grafana/grafana-plugin-repository/projects/2).

**Comment by Tamirk** [comment]
@marcusolsson What is the expected ETA for a review on this?

**Comment by marcusolsson** [comment]
Hi @Tamirk,

When reviewing this, I noticed that you're collecting (anonymous) usage data. As part of our guidelines, we don't allow plugins to track any data from our users, even if it's anonymous. 

I understand that this is something that hasn't been added in this release. Unfortunately, it seems like we missed this when reviewing your plugin in the past.

We'd like to kindly ask that you remove any usage tracking from your plugin, and submit it again. Please let me know if you have any questions.

**Comment by LiorRacer** [comment]
Hi @marcusolsson, 

I removed the mixpanel integration from the plugin

**Comment by Tamirk** [comment]
@marcusolsson appreciate if you can advise when you think this can be review and hopefully approved

**Comment by marcusolsson** [comment]
I'm reviewing this during the day. Though it seems like there are some issues with how the plugin is packaged.

[Plugin validator](https://grafana-plugins-web-vgmmyppaka-lz.a.run.app/?url=https://github.com/doitintl/bigquery-grafana/releases/download/2.0.1/doitintl-bigquery-datasource-2.0.1.zip)

You can read more about how to package plugins on [Package a plugin](https://grafana.com/docs/grafana/latest/developers/plugins/package-a-plugin), but essentially you'd need to something like this:

```
mv dist/ doitintl-bigquery-datasource
zip dointl-bigquery-datasource-2.0.1.zip doitintl-bigquery-datasource -r
```

**Comment by marcusolsson** [comment]
I built the plugin from the 2.0.1 branch and noticed that the mixpanel package is still there. As I'm not sure what to test, I'm holding off on this for now until you've had a chance to look into it.

**Comment by marcusolsson** [comment]
I've built and reviewed the 2.0.1 branch, which looks good. I'll await the repackaged plugin and to a quick review on the changes, but after that it should be good to go.

**Comment by Tamirk** [comment]
@marcusolsson Great news thank you for that. We'll update the package soon per your instructions

**Comment by LiorRacer** [comment]
Hi @marcusolsson 
I removed the package from package.json and repacked the plugin

**Comment by marcusolsson** [comment]
Awesome! Could you update the PR with the new URL and MD5 to point to the new release?

**Comment by LiorRacer** [comment]
Done

File: repo.json [hunk 1]
```diff
@@ -3607,6 +3607,21 @@
...
+          "version": "2.0.0",
+          "commit": "adc3461dfcae7fd2b92cb1748f585ee4f5cbfe72",
+          "url": "https://github.com/doitintl/bigquery-grafana"
+        },
+        {
+          "version": "2.0.1",
+          "url": "https://github.com/doitintl/bigquery-grafana",
+          "download": {
+            "any": {
+              "url": "https://github.com/doitintl/bigquery-grafana/releases/download/2.0.1/dointl-bigquery-datasource-2.0.1.zip",
```
Comment (marcusolsson) [inline repo.json:3621]:
```suggestion
              "url": "https://github.com/doitintl/bigquery-grafana/releases/download/2.0.1/doitintl-bigquery-datasource-2.0.1.zip",
```

File: repo.json [hunk 2]
```diff
@@ -3607,6 +3607,21 @@
           "version": "1.0.8",
           "commit": "d244fe87186d4ec2bffbe3fe0fb1c331fc63b60e",
           "url": "https://github.com/doitintl/bigquery-grafana"
+        },
+        {
+          "version": "2.0.0",
```
Comment (marcusolsson) [inline repo.json:3612]: This is not a valid release, would you mind removing it from the PR?

**Review by marcusolsson — CHANGES_REQUESTED** [review]

File: repo.json (hunk 2 above)
Comment (LiorRacer) [reply repo.json]: No problem

**Review by marcusolsson — APPROVED** [review]

**Comment by marcusolsson** [comment]
Beautiful! 🚀 

CleanShot 2021-02-15 at 18 48 10@2x

I've [published](https://grafana.com/grafana/plugins/doitintl-bigquery-datasource?version=2.0.1) the new version! 🎉

**Comment by Tamirk** [comment]
Amazing news!!! Thank you so much

Best,
Tamir Klein | Director Of Engineering // DoiT International
/ +972-54-8119220

**Pull request merged**